### PR TITLE
Enable lint rules for React

### PR DESCRIPTION
**Pull request by tnorling**
Enables a bunch of lint rules for React

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -52,6 +53,24 @@
...
-        "spaced-comment": 2
+        "spaced-comment": 2,
+        "react/jsx-closing-bracket-location": 2,
+        "react/jsx-closing-tag-location": 2,
+        "react/jsx-curly-spacing": [2, {"when": "always"}],
+        "react/jsx-filename-extension": [2, {"extensions": [".jsx", ".tsx"], "allow": "as-needed"}],
+        "react/jsx-pascal-case": 2,
+        "react/jsx-tag-spacing": 2,
+        "react/jsx-wrap-multilines": 2,
+        "react/no-multi-comp": 2,
```
Comment (jasonnutter) [inline .eslintrc.json:64]: The most controversial lint rule of all time lol.

Comment (jasonnutter) [reply .eslintrc.json:64]: (I'm a fan though)

File: lib/msal-react/src/MsalProvider.tsx [hunk 2]
```diff
@@ -77,7 +77,7 @@ export function MsalProvider({instance, children}: MsalProviderProps): React.Rea
     };
 
     return (
-        <MsalContext.Provider value={contextValue}>
+        <MsalContext.Provider value={ contextValue }>
```
Comment (jasonnutter) [inline lib/msal-react/src/MsalProvider.tsx:80]: Hmm, admittedly not the biggest of this.

Comment (tnorling) [reply lib/msal-react/src/MsalProvider.tsx:80]: Fair enough, do you prefer no spaces or no rule?

Comment (jasonnutter) [reply lib/msal-react/src/MsalProvider.tsx:80]: I think I like this?

`{ "when": "never", "attributes": { "allowMultiline": false }, "children": true }`

Comment (tnorling) [reply lib/msal-react/src/MsalProvider.tsx]: Done

**Review by jasonnutter — APPROVED** [review]

**Review by jo-arroyo — APPROVED** [review]

**Pull request merged**